oauthenticator/github.py

```diff
@@ -116,7 +116,15 @@ def authenticate(self, handler, data=None):
         resp = yield http_client.fetch(req)
         resp_json = json.loads(resp.body.decode('utf8', 'replace'))
 
-        access_token = resp_json['access_token']
+        if 'access_token' in resp_json:
+            access_token = resp_json['access_token']
+        elif 'error_description' in resp_json:
+            raise HTTPError(403,
+                "An access token was not returned: {}".format(
```

Comment:
Should this be a 403? I think typically this will be a result of a bug, so maybe leave it as a 500 with the error_description?

Reply:
Well it’s  not necessarily a bug - if the auth code code out of sync (eg someone left a browser cache open for a long time) then a simple retry will fix it. Happy to make it a 500 if that’s more consitent with the jupyterhub style.